lib/grape/error_formatter/base.rb

```diff
@@ -3,9 +3,13 @@ module ErrorFormatter
     module Base
       def present(message, env)
         present_options = {}
-        present_options[:with] = message.delete(:with) if message.is_a?(Hash)
+        presented_message = message
+        if presented_message.is_a?(Hash) && presented_message.key?(:with)
+          present_options[:with] = presented_message[:with]
+          presented_message = presented_message.reject { |k, _| k == :with }
+        end
```

Comment:
The body of the `if` above could have been:
```ruby
presented_message = presented_message.dup
present_options[:with] = presented_message.delete(:with)
```
I'm on the fence about which would be clearer.